src/vos/vos_common.c

```diff
@@ -671,17 +671,20 @@ vos_self_nvme_fini(void)
 }
 
 /* Storage path, NVMe config & numa node used by standalone VOS */
-#define VOS_NVME_CONF		"/etc/daos_nvme.conf"
+#define VOS_NVME_CONF		"daos_nvme.conf"
 #define VOS_NVME_NUMA_NODE	DAOS_NVME_NUMANODE_NONE
 #define VOS_NVME_MEM_SIZE	1024
 #define VOS_NVME_HUGEPAGE_SIZE	2	/* 2MB */
 #define VOS_NVME_NR_TARGET	1
 
 static int
-vos_self_nvme_init()
+vos_self_nvme_init(const char *vos_path)
 {
-	int rc;
-	int fd;
+	char	nvme_conf[128];
+	int	rc, fd;
+
+	D_ASSERT(vos_path != NULL);
+	snprintf(nvme_conf, sizeof(nvme_conf), "%s/%s", vos_path, VOS_NVME_CONF);
```

Comment:
You need to use D_ASPRINTF()/D_FREE here, snprintf with a fixed size buffer isn't even remotely safe.

Reply:
These issues should be fixed on master branch first

Reply:
There are two issues here.

One is that snprintf does not write null byte if the source string is longer.  In other places, we often "handle" this by always writing the last byte to 0.

But Ashley's proposal for using D_ASPRINTF is preferrable since it won't truncate the string at all.